apps/emqx_web_hook/src/emqx_web_hook_actions.erl

```diff
@@ -335,8 +337,26 @@ method(POST) when POST == <<"POST">>; POST == <<"post">> -> post;
 method(PUT) when PUT == <<"PUT">>; PUT == <<"put">> -> put;
 method(DEL) when DEL == <<"DELETE">>; DEL == <<"delete">> -> delete.
 
-headers(Headers) ->
-    emqx_http_lib:normalise_headers(maps:to_list(Headers)).
+preproc_and_normalise_headers(Headers) ->
```

Comment:
return `{tmpl_tokens, Headers :: map()}` or `Headers :: map()` from this function,
so the 99% of the chance, there is no need for template rendering later at runtime.